index.d.ts

```diff
@@ -0,0 +1,15 @@
+import * as React from "react";
```

Comment:
I have started reviewing the PR. Will create a test case with codesandbox as well.

First thing to ask in advance: isn't the presence of `index.d.ts` in place of `PropTypes` assumes we need to be in TypeScript environment?